internal/cache/sharedcache.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2025 The argocd-agent Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/argoproj/argo-cd/v3/common"
+	cacheutil "github.com/argoproj/argo-cd/v3/util/cache"
+	appstatecache "github.com/argoproj/argo-cd/v3/util/cache/appstate"
+	"github.com/redis/go-redis/v9"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	SourceAgent     = "agent"
+	SourcePrincipal = "principal"
+)
+
+type SharedCache struct {
+	mu                     sync.RWMutex
+	agentCacheInstance     *appstatecache.Cache
+	principalCacheInstance *appstatecache.Cache
+}
+
+// getAgentCacheInstance returns the agent cache instance
+func (sc *SharedCache) getAgentCacheInstance(ctx context.Context, kubeclient kubernetes.Interface,
```

Comment:
Thinking out loud, do we even need this `SharedCache` file/package at all? Why not just put `cacheInstance *appstatecache.Cache` in all the places where we are currently adding a reference to SharedCache?

e.g. remove SharedCache from `type Agent struct {` and in `type Manager struct {`, and just add appstatecache.Cache